lib/foreman_scap_client/client.rb

```diff
@@ -111,5 +100,43 @@ def foreman_proxy_uri
       foreman_proxy_port = config[:port]
       "https://#{foreman_proxy_fqdn}:#{foreman_proxy_port}"
     end
+
+    def generate_https_object(uri)
+      https = Net::HTTP.new(uri.host, uri.port)
+      https.use_ssl = true
+      https.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      https.ca_file = config[:ca_file]
+      begin
+        https.cert = OpenSSL::X509::Certificate.new File.read(config[:host_certificate])
+        https.key = OpenSSL::PKey::RSA.new File.read(config[:host_private_key])
+      rescue StandardError => e
+        puts 'Unable to load certs'
+        puts e.message
+        exit(3)
+      end
+      https
+    end
+
+    def ensure_scan_file
+      return if File.exist?(config[@policy_id][:content_path])
+      puts "File #{config[@policy_id][:content_path]} is missing. Downloading it from proxy"
+      begin
+        FileUtils.mkdir_p(File.dirname(config[@policy_id][:content_path]))
+        uri = URI.parse(download_uri(config[@policy_id][:download_path]))
+        puts "Download scap content xml from: #{uri}"
+        request = generate_https_object(uri).get(uri)
```

Comment:
while testing, I'm getting this
SCAP file is missing and download failed with error: undefined method `empty?' for #<URI::HTTPS:0x7fdb467efb98>

when I add trace I can see:
```
/usr/lib/ruby/1.8/net/http.rb:1478:in `initialize'
/usr/lib/ruby/1.8/net/http.rb:1596:in `initialize'
/usr/lib/ruby/1.8/net/http.rb:772:in `new'
/usr/lib/ruby/1.8/net/http.rb:772:in `get'
/usr/lib/ruby/gems/1.8/gems/foreman_scap_client-0.1.0/lib/foreman_scap_client/client.rb:131:in `ensure_scan_file'
/usr/lib/ruby/gems/1.8/gems/foreman_scap_client-0.1.0/lib/foreman_scap_client/client.rb:14:in `run'
/usr/lib/ruby/gems/1.8/gems/foreman_scap_client-0.1.0/bin/foreman_scap_client:9
/usr/bin/foreman_scap_client:19:in `load'
/usr/bin/foreman_scap_client:19
```

changing this line to `generate_https_object(uri).get(uri.path)` seems to fix it for me
(tested on Ruby 1.8.7)